docs/development/contributor_guidelines.rst

```diff
@@ -107,6 +107,86 @@ Run the checks locally via:
 
       make yamllint
 
+Type hints in Python code
+-------------------------
+
+By adding type hints/annotations in the Python code, we are making the code-base 
+easier to maintain in the long-run by explicitly specifying the expected input/output
```

Comment:
Nit: "expected input/output of various functions" -> "expected input/output types of various functions"